src/backends/perf_pt/collect.c

```diff
@@ -142,6 +163,84 @@ bool perf_pt_start_tracer(struct tracer_ctx *, struct perf_pt_trace *, struct pe
 bool perf_pt_stop_tracer(struct tracer_ctx *tr_ctx, struct perf_pt_cerror *);
 bool perf_pt_free_tracer(struct tracer_ctx *tr_ctx, struct perf_pt_cerror *);
 
+
+/*
+ * Reads samples from the data buffer and acts upon them appropriately.
```

Comment:
This sentence almost doesn't say anything. Can we be a bit more specific?

Reply:
It's a bit like the interpreter loop in a VM. It tells us what actions to perform. Did you have something specific in mind?

Reply:
I didn't really understand what the function does so... no ;)

Reply:
OK, I'll try to find something more explicit.